scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/RecurrenceGoal.java

```diff
@@ -173,31 +154,31 @@ protected RecurrenceGoal() { }
       Duration start, Duration end, Plan plan)
   {
     final var conflicts = new java.util.LinkedList<MissingActivityConflict>();
-
-    var goalConstraint = this.stateConstraints;
-
     //determine how much flexibility there is in creating activities
     final var recurrenceFlexibility = recurrenceInterval.end.minus(
         recurrenceInterval.start);
 
+    if(end.minus(start).noLongerThan(recurrenceInterval.end)){
+      return conflicts;
+    }
+
     //walk forward in time by full allowed stride lengths
     for (var intervalT = start.plus(recurrenceInterval.end);
-         intervalT.compareTo(end) < 0;
-         intervalT = intervalT.plus(recurrenceInterval.end)
+         ;intervalT = intervalT.plus(recurrenceInterval.end)
     ) {
       //REVIEW: technically, could create activity at extremely short
       //        intervals (ie minT=0) and still satisfy the goal as currently
       //        framed, which is exactly what the current solver will do since
       //        it chooses the minimum. but it looks ugly. so for now passing
       //        a limited flexibility for creation
       final var minT = intervalT.minus(recurrenceFlexibility);
-      var interval =  new Windows(Window.between(minT,intervalT));
-      if(goalConstraint != null) {
-        interval.intersectWith(goalConstraint.findWindows(plan, new Windows(Window.between(start, end))));
-      }
+      var interval =  new Windows(Window.between(intervalT.minus(recurrenceInterval.end), Duration.min(intervalT, end)));
       conflicts.add(new MissingActivityTemplateConflict(
           this, new Windows(
           interval),this.getActTemplate()));
+      if(intervalT.compareTo(end) > 0){
+        break;
+      }
```

Comment:
Maybe you could wrap this in a do-while loop instead of a for loop:

```
    var intervalT = start.plus(recurrenceInterval.end);
    do{
      //REVIEW: technically, could create activity at extremely short
      //        intervals (ie minT=0) and still satisfy the goal as currently
      //        framed, which is exactly what the current solver will do since
      //        it chooses the minimum. but it looks ugly. so for now passing
      //        a limited flexibility for creation
      final var minT = intervalT.minus(recurrenceFlexibility);
      var interval =  new Windows(Window.between(intervalT.minus(recurrenceInterval.end), Duration.min(intervalT, end)));
      conflicts.add(new MissingActivityTemplateConflict(
          this, new Windows(
          interval),this.getActTemplate()));
      intervalT = intervalT.plus(recurrenceInterval.end)
   
    } while(intervalT.compareTo(end) <= 0);
```

Also, minT isn't being used in the loop

` final var minT = intervalT.minus(recurrenceFlexibility);`

Reply:
I've done what you suggested.

Reply:
Actually, I rolled back on these changes because the number of conflicts produced is different. Let's take an example with `start` = 0, `end` = 2h10 and a `recurrenceInterval` of 1 hour. 
`intervalT` = 0 + 1h = 1h, we insert a conflict at 1h
`intervalT` = 1h + 1h = 2h, we insert a conflict at 2h
`intervalT` = 2h+1h = 3h, `intervalT` is now after the end but nonetheless we need to create a conflict for the period [2h, 2h10]. We ensure that the temporal context of the conflict does not go beyond end (with the `Duration.min` in line 175). 

Your changes would change this flow and exit the loop when we increment intervalT to 3h before creating the appropriate conflict.